Frends.Community.RabbitMQ/RabbitMQTask.cs

```diff
@@ -12,88 +12,196 @@ namespace Frends.Community.RabbitMQ
-        /// Writes message to a queue
+        /// Writes messages into a queue with simple publish. Message is not under transaction. No rollback of failed messages.  
         /// </summary>
         /// <param name="inputParams"></param>
-        public static bool WriteMessage([PropertyTab]WriteInputParams inputParams)
+        public static bool WriteMessage([PropertyTab] WriteInputParams inputParams)
         {
-            OpenConnectionIfClosed(inputParams.HostName, inputParams.ConnectWithURI);
-
-            if (inputParams.Create)
+            IConnection connection = CreateConnection(inputParams.HostName, inputParams.ConnectWithURI);
+            IModel channel = connection.CreateModel();
+            try
             {
-                _channel.QueueDeclare(queue: inputParams.QueueName,
-                                durable: inputParams.Durable,
-                                exclusive: false,
-                                autoDelete: false,
-                                arguments: null);
-
-            }
+                if (inputParams.Create)
+                {
+                    channel.QueueDeclare(queue: inputParams.QueueName,
+                        durable: inputParams.Durable,
+                        exclusive: false,
+                        autoDelete: false,
+                        arguments: null);
+                    channel.ConfirmSelect();
+                }
 
-            IBasicProperties basicProperties = null;
-            if (inputParams.Durable == true)
-            {
+                IBasicProperties basicProperties = null;
 
-                basicProperties = _channel.CreateBasicProperties();
-                basicProperties.Persistent = true;
+                if (inputParams.Durable == true)
+                {
+                    basicProperties = channel.CreateBasicProperties();
+                    basicProperties.Persistent = true;
+                }
+                
+                if (inputParams.WriteMessageCount <= 1)
+                {
+                    channel.BasicPublish(exchange:
+                        inputParams.ExchangeName,
+                        routingKey: inputParams.RoutingKey,
+                        basicProperties: basicProperties,
+                        body: inputParams.Data);
+                    return true;
+                }
 
+                return false;
+            }
+            finally
+            {
+               CloseConnection(channel, connection); 
             }
+        }
+        
+        /// <summary>
+        /// Writes messages into a queue with batch publish. All messages are under transaction. If one the message failes all messages will be rolled back.  
+        /// </summary>
+        /// <param name="inputParams"></param>
+        public static bool WriteBatchMessage([PropertyTab] WriteInputParams inputParams)
+        {
+            IConnection connection = CreateConnection(inputParams.HostName, inputParams.ConnectWithURI);
```

Comment:
The connection seems to be recreated for each message sent to the batch. This seems a bit odd for transactional message sending, does it really work?

Reply:
Creates now a connection for both simple messaging and batch messaging (with batch shares a connection between all of the batch components and finally closes the batch specific connection).